session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSession.scala

```diff
@@ -51,7 +51,8 @@ final class CassandraSession(
     executionContext: ExecutionContext,
     log: LoggingAdapter,
     metricsCategory: String,
-    init: CqlSession => Future[Done])
+    init: CqlSession => Future[Done],
+    onClose: () => Unit)
     extends NoSerializationVerificationNeeded {
```

Comment:
the session is coupled to the ActorSystem so one question is if it should automatically close when the ActorSystem has been terminated?

Reply:
that would also solve the problem with blocking close from the Journal actors, since they would not be responsible for closing